bodhi/server/services/releases.py

```diff
@@ -327,3 +346,55 @@ def save_release(request):
     request.db.flush()
 
     return r
+
+
+@critical_packages.put(schema=bodhi.server.schemas.UpdateCriticalPackages,
+                       renderer='json',
+                       error_handler=bodhi.server.services.errors.json_handler,
+                       validators=(colander_body_validator, validate_tags, validate_enums)
+                       )
+def add_packages(request):
+    """
+    Add a list of critical path packages to a specific release.
+
+    Args:
+        request (pyramid.request): The current request.
+    Returns:
+        dict: The dictionary {'status': 'success'}.
+    """
+    data = request.validated
+    pkg_list = data.pop('packages', None)
+
+    id = request.matchdict.get('name')
+    release = Release.get(id)
+
+    session = Session()
+    release.add_critpath_pkgs(pkg_list, session)
+
+    return dict(status=u'success', packages=release.critpath_pkgs)
+
+
+@critical_packages.delete(schema=bodhi.server.schemas.UpdateCriticalPackages,
+                          permission='edit', renderer='json',
+                          error_handler=bodhi.server.services.errors.json_handler,
+                          validators=(colander_body_validator, validate_tags, validate_enums)
+                          )
+def remove_packages(request):
+    """
+    Remove a list of critical path packages from a specific release.
+
+    Args:
+        request (pyramid.request): The current request.
+    Returns:
+        dict: The dictionary {'status': 'success'}.
+    """
+    data = request.validated
+    pkg_list = data.pop('packages', None)
+
+    id = request.matchdict.get('name')
+    release = Release.get(id)
```

Comment:
Similarly here, we should use validators to ensure that pkg_list is already Package objects and release should already be validated as an object as well.